diff -ru orig.binutils-2.27/ld/scripttempl/elf.sc binutils-2.27/ld/scripttempl/elf.sc
--- orig.binutils-2.27/ld/scripttempl/elf.sc	2016-08-03 04:36:54.000000000 -0300
+++ binutils-2.27/ld/scripttempl/elf.sc	2016-08-13 16:49:51.306136428 -0300
@@ -554,6 +554,9 @@
   ${CREATE_SHLIB+${RELOCATING+. = ${SHLIB_DATA_ADDR-${DATA_SEGMENT_ALIGN}};}}
   ${CREATE_PIE+${RELOCATING+. = ${SHLIB_DATA_ADDR-${DATA_SEGMENT_ALIGN}};}}
 
+  /* Hacky hackity hack. Need this because otherwise data segment is not aligned enough. */
+  . = ALIGN(0x10000);
+
   /* Exception handling  */
   .eh_frame     ${RELOCATING-0} : ONLY_IF_RW { KEEP (*(.eh_frame)) ${RELOCATING+*(.eh_frame.*)} }
   .gnu_extab    ${RELOCATING-0} : ONLY_IF_RW { *(.gnu_extab) }
