diff -ru orig.SDL-Vita/src/render/vita/SDL_render_vita.c SDL-Vita/src/render/vita/SDL_render_vita.c
--- orig.SDL-Vita/src/render/vita/SDL_render_vita.c	2016-11-17 12:01:51.378374758 -0400
+++ SDL-Vita/src/render/vita/SDL_render_vita.c	2016-11-17 12:02:23.094678099 -0400
@@ -28,7 +28,6 @@
 #include <psp2/types.h>
 #include <psp2/display.h>
 #include <psp2/gxm.h>
-#include <psp2/moduleinfo.h>
 #include <psp2/kernel/processmgr.h>
 #include <stdio.h>
 #include <string.h>
